<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TuringMachineClass</class>
 <widget class="QMainWindow" name="TuringMachineClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1159</width>
    <height>649</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1159</width>
    <height>649</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1159</width>
    <height>649</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>TuringMachine</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>1141</width>
        <height>131</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>1141</width>
        <height>131</height>
       </size>
      </property>
      <property name="title">
       <string>Лента</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QTableWidget" name="tape">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>60</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>60</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Cooper Black</family>
           <pointsize>13</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::DoubleClicked</set>
         </property>
         <property name="textElideMode">
          <enum>Qt::ElideMiddle</enum>
         </property>
         <attribute name="horizontalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderCascadingSectionResizes">
          <bool>true</bool>
         </attribute>
         <attribute name="horizontalHeaderDefaultSectionSize">
          <number>50</number>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderCascadingSectionResizes">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderDefaultSectionSize">
          <number>50</number>
         </attribute>
        </widget>
       </item>
       <item row="1" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="step">
           <property name="minimumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Шаг</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="begin">
           <property name="minimumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Начать</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="stop">
           <property name="minimumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Остановить</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>718</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="exportTape">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Экспорт</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="importTape">
           <property name="minimumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Импорт</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="minimumSize">
       <size>
        <width>1141</width>
        <height>451</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>1141</width>
        <height>451</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="title">
       <string>Устройство управления</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0" rowspan="3" colspan="2">
        <widget class="QTableWidget" name="controller">
         <property name="minimumSize">
          <size>
           <width>581</width>
           <height>371</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>581</width>
           <height>371</height>
          </size>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::DoubleClicked</set>
         </property>
         <attribute name="horizontalHeaderCascadingSectionResizes">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderStretchLastSection">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QFrame" name="frame_2">
         <property name="minimumSize">
          <size>
           <width>529</width>
           <height>46</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>529</width>
           <height>46</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::WinPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Следующая команда:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="nextCommand">
            <property name="font">
             <font>
              <pointsize>15</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>-</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QFrame" name="frame">
         <property name="minimumSize">
          <size>
           <width>529</width>
           <height>46</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>529</width>
           <height>46</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::WinPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Текущее состояние:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="currentState">
            <property name="font">
             <font>
              <pointsize>15</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>-</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QPushButton" name="exportController">
           <property name="minimumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="text">
            <string>Экспорт</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="importController">
           <property name="minimumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>75</width>
             <height>23</height>
            </size>
           </property>
           <property name="text">
            <string>Импорт</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="1">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>398</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="2">
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Редактирование</string>
         </property>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>30</y>
            <width>501</width>
            <height>85</height>
           </rect>
          </property>
          <property name="title">
           <string>Добавить состояние</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="minimumSize">
                <size>
                 <width>120</width>
                 <height>23</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>120</width>
                 <height>23</height>
                </size>
               </property>
               <property name="text">
                <string>Введите состояние:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="newState">
               <property name="minimumSize">
                <size>
                 <width>272</width>
                 <height>20</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>272</width>
                 <height>20</height>
                </size>
               </property>
               <property name="maxLength">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="addNewState">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="text">
                <string>Добавить</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QCheckBox" name="beginState">
               <property name="text">
                <string>Начальное</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="endState">
               <property name="text">
                <string>Конечное</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="currState">
               <property name="text">
                <string>Текущее</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>120</y>
            <width>501</width>
            <height>83</height>
           </rect>
          </property>
          <property name="title">
           <string>Добавить символ алфавита</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLabel" name="label_4">
               <property name="minimumSize">
                <size>
                 <width>120</width>
                 <height>23</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>120</width>
                 <height>23</height>
                </size>
               </property>
               <property name="text">
                <string>Введите символ:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="newSymbol">
               <property name="minimumSize">
                <size>
                 <width>272</width>
                 <height>20</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>272</width>
                 <height>20</height>
                </size>
               </property>
               <property name="maxLength">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="addNewSymbol">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="text">
                <string>Добавить</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="emptySymbol">
             <property name="text">
              <string>Пустой символ</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1159</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>step</tabstop>
  <tabstop>begin</tabstop>
  <tabstop>stop</tabstop>
  <tabstop>exportTape</tabstop>
  <tabstop>importTape</tabstop>
  <tabstop>tape</tabstop>
  <tabstop>controller</tabstop>
  <tabstop>exportController</tabstop>
  <tabstop>importController</tabstop>
 </tabstops>
 <resources>
  <include location="turingmachine.qrc"/>
 </resources>
 <connections/>
</ui>
